### PR TITLE
Add queued store and display active users

### PR DESCRIPTION
**Pull request by jmthomas**

### CODE REVIEW

File: openc3/lib/openc3/utilities/store_queued.rb [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+# This file may also be used under the terms of a commercial license
+# if purchased from OpenC3, Inc.
+
+require 'openc3/utilities/store'
+
+module OpenC3
+  class StoreQueued
+    # How often in seconds the store is updated
+    UPDATE_INTERVAL = 2
```
Comment (ryanmelt) [inline openc3/lib/openc3/utilities/store_queued.rb:24]: why 2?

File: openc3/lib/openc3/utilities/store_queued.rb [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+    # Variable that holds the singleton instance
+    @instance = nil
+
+    # Mutex used to ensure that only one instance is created
+    @@instance_mutex = Mutex.new
+
+    # Thread used to post metrics across all classes
```
Comment (ryanmelt) [inline openc3/lib/openc3/utilities/store_queued.rb:32]: metrics?

File: openc3/lib/openc3/utilities/store_queued.rb [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    def self.instance()
+      return @instance if @instance
+
+      @@instance_mutex.synchronize do
+        @instance ||= self.new()
+        return @instance
+      end
+    end
+
+    def self.shutdown
```
Comment (ryanmelt) [inline openc3/lib/openc3/utilities/store_queued.rb:48]: who calls this?

**Review by ryanmelt — CHANGES_REQUESTED** [review]

File: openc3/lib/openc3/utilities/store_queued.rb (hunk 1 above)
Comment (ryanmelt) [reply openc3/lib/openc3/utilities/store_queued.rb:24]: This needs to be an argument to the instance on first use (like System).   I want to use this same class to implement #987

File: openc3/lib/openc3/utilities/store_queued.rb (hunk 2 above)
Comment (jmthomas) [reply openc3/lib/openc3/utilities/store_queued.rb:32]: typo

File: openc3/lib/openc3/utilities/store_queued.rb (hunk 3 above)
Comment (jmthomas) [reply openc3/lib/openc3/utilities/store_queued.rb:48]: no one ... it's left over from the metrics thread I borrowed

Comment (ryanmelt) [reply openc3/lib/openc3/utilities/store_queued.rb:48]: Should be added to at_exit when instance created

File: openc3/lib/openc3/utilities/store_queued.rb [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+module OpenC3
+  class StoreQueued
+    # Variable that holds the singleton instance
+    @instance = nil
+
+    # Mutex used to ensure that only one instance is created
+    @@instance_mutex = Mutex.new
+
+    # Thread used to call methods on the store
+    @@update_thread = nil
```
Comment (ryanmelt) [inline openc3/lib/openc3/utilities/store_queued.rb:30]: This should just be an instance variable

File: openc3/lib/openc3/utilities/store_queued.rb [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+    # Sleeper used to delay update thread
+    @@update_sleeper = nil
```
Comment (ryanmelt) [inline openc3/lib/openc3/utilities/store_queued.rb:33]: This should just be an instance variable

File: openc3/lib/openc3/utilities/store_queued.rb (hunk 3 above)
Comment (ryanmelt) [reply openc3/lib/openc3/utilities/store_queued.rb:46]: Should be instance method

**Review by ryanmelt — CHANGES_REQUESTED** [review]

**Comment by ryanmelt** [comment]
Let me make some changes to this one before you merge it.

**Pull request merged**